core/src/main/java/com/amplifyframework/devmenu/LogEntry.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.devmenu;
+
+import android.util.Log;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.amplifyframework.logging.LogLevel;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Objects;
+
+/**
+ * A representation of a log.
+ */
+public final class LogEntry {
```

Comment:
Ah crap, actually. In my last round of feedback, I suggested you do:
```java
Collection.sort(listOfLogEntry, (left, right) -> { /* blah balh */ });
```
With the comparator specified at the call site.

However, cleaner and better encapsulated would be to have `public final class LogEntry implements Comparable<LogEntry> {`. Then, you would only need to do `Collections.sort(listOfLogEntry)`, and it would Just Work™.